src/cargo/core/resolver/encode.rs

```diff
@@ -338,7 +338,7 @@ impl EncodableResolve {
         let mut to_remove = Vec::new();
         for (k, v) in metadata.iter().filter(|p| p.0.starts_with(prefix)) {
             to_remove.push(k.to_string());
-            let k = &k[prefix.len()..];
+            let k = k.strip_prefix(prefix).unwrap();
```

Comment:
nit: you could turn the `filter` into a `filter_map` but then each iteration would have two different keys so unsure if its worth it